### PR TITLE
Execute status rule if every rule fails

### PR DESCRIPTION
**Pull request by andrecrts**
**Issue Reference**:[#530](https://github.com/gagoar/use-herald-action/issues/530)

## Description
i updated the getMatchingRules method to return all the status rules and also the validation to update the status to be success or failure depending on the match result and not if it exists on the Matching rules

## Motivation and Context
it fixes this [Issue 530](https://github.com/gagoar/use-herald-action/issues/530)

## How Has This Been Tested?
i've tested it on my private repo where i have just one status rule

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -132,7 +132,7 @@ export const main = async (): Promise<void> => {
               if (!rulesForAction.length) {
                 return promises;
               }
-
+              console.log(actionName)
```
Comment (gagoar) [inline src/index.ts:135]: Remove this please.

File: __mocks__/rules/rule3.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "bot validation",
+    "description": "bot validation",
+    "action": "status",
+    "eventJsonPath": [
+        "$[?(@.type.match(/bot/i))]"
+    ],
+    "targetURL": "https://github.com/gagoar/use-herald-action/wiki"
+}
```
Comment (gagoar) [inline __mocks__/rules/rule3.json:9]: line jump.

File: src/rules.ts [hunk 3]
```diff
@@ -268,7 +268,7 @@ class MatchingRules extends Array<MatchingRule> {
 
     const matchedRules = await Promise.all(matchingRules);
 
-    const filtered = matchedRules.filter((rule) => isMatchingRule(rule) && rule.matched);
+    const filtered = matchedRules.filter((rule) => isMatchingRule(rule) && rule.matched || rule.action === RuleActions.status);
```
Comment (gagoar) [inline src/rules.ts:271]: This should be a group to future proof the concept. 



```suggestion
const alwaysMatchedRulesTypes = [RuleActions.status];
const filtered = matchedRules.filter((rule) =>  alwaysMatchedRulesTypes.includes(rule.action) || isMatchingRule(rule) && rule.matched);
```

Comment (andrecrts) [reply src/rules.ts]: you are right, i did the change

File: src/rules.ts [hunk 4]
```diff
@@ -268,8 +268,11 @@ class MatchingRules extends Array<MatchingRule> {
 
     const matchedRules = await Promise.all(matchingRules);
 
-    const filtered = matchedRules.filter((rule) => isMatchingRule(rule) && rule.matched);
+    const alwaysMatchedRulesTypes = [RuleActions.status]
 
+    const filtered = matchedRules.filter((rule) =>
+      alwaysMatchedRulesTypes.includes(rule.action as RuleActions) || isMatchingRule(rule) && rule.matched);
+    console.log({ filtered })
```
Comment (gagoar) [inline src/rules.ts:275]: Please use `debug` to output these type of things.

**Review by gagoar — CHANGES_REQUESTED** [review]
still a console.log left.

File: __tests__/index.spec.ts [hunk 5]
```diff
@@ -37,6 +37,10 @@ const getGithubMock = () =>
   });
 
 describe('use-herald-action', () => {
+  const owner = 'gagoar';
+  const repo = 'example_repo';
+  const sha = 'ec26c3e57ca3a959ca5aad62de7213c562f8c821';
+
```
Comment (gagoar) [inline __tests__/index.spec.ts:43]: why do you need these available here and not within the beforeEach? are these used elsewhere?

File: src/rules.ts [hunk 6]
```diff
@@ -268,8 +268,10 @@ class MatchingRules extends Array<MatchingRule> {
 
     const matchedRules = await Promise.all(matchingRules);
 
-    const filtered = matchedRules.filter((rule) => isMatchingRule(rule) && rule.matched);
+    const alwaysMatchedRulesTypes = [RuleActions.status]
```
Comment (gagoar) [inline src/rules.ts:271]: this const name is causing me issues. 

Can't come up with a better name for it. maybe @cyamonide can help?

File: __tests__/index.spec.ts (hunk 5 above)
Comment (andrecrts) [reply __tests__/index.spec.ts:43]: because i need them on two tests, so i don't need to declare them twice on each test

**Review by gagoar — APPROVED** [review]
We will wait till @cyamonide says something tomorrow in search of a better naming for `alwaysMatchRuleType` if not we will merge as is.

File: src/rules.ts (hunk 6 above)
Comment (cyamonide) [reply src/rules.ts]: instead of _always matched_, how about _mandatory_ or _required_? first name that came to mind was `mandatoryRulesTypes`.

Comment (gagoar) [reply src/rules.ts]: Yea. @andrecrts mind changing it?

Comment (andrecrts) [reply src/rules.ts]: changed

**Comment by gagoar** [comment]
Thanks @andrecrts I will release shortly

**Pull request merged**